overlord/snapstate/snapstate.go

```diff
@@ -2389,6 +2389,13 @@ func UpdateWithDeviceContext(st *state.State, name string, opts *RevisionOptions
 			addPrereq(prqt, info)
 			toUpdate = append(toUpdate, installSnapInfo{info})
 		case store.ErrNoUpdateAvailable:
+			// if there isn't an update available, then we should sill add the
+			// current info to the prereq tracker
+			currentInfo, err := snapst.CurrentInfo()
```

Comment:
Not really sure what to do with this error, since we can't return it here, since the infoErr is introspected later on.